src/ngraph/runtime/interpreter/int_backend.hpp

```diff
@@ -408,115 +427,104 @@ class ngraph::runtime::interpreter::INTBackend : public Backend
         case OP_TYPEID::Broadcast:
         {
             const op::Broadcast* broadcast = static_cast<const op::Broadcast*>(&node);
-            Shape in_shape = args[0]->get_shape();
-            Shape out_shape = out[0]->get_shape();
+            Shape in_shape = node.get_input_shape(0);
+            Shape out_shape = node.get_output_shape(0);
             AxisSet broadcast_axes = broadcast->get_broadcast_axes();
-            reference::broadcast<T>(args[0]->get_data_ptr<T>(),
-                                    out[0]->get_data_ptr<T>(),
+            reference::broadcast<T>(static_cast<const T*>(args[0]),
+                                    static_cast<T*>(out[0]),
                                     in_shape,
                                     out_shape,
                                     broadcast_axes);
             break;
         }
         case OP_TYPEID::Ceiling:
         {
+            size_t element_count = shape_size(node.get_output_shape(0));
             reference::ceiling<T>(
-                args[0]->get_data_ptr<T>(), out[0]->get_data_ptr<T>(), out[0]->get_element_count());
+                static_cast<const T*>(args[0]), static_cast<T*>(out[0]), element_count);
             break;
         }
         case OP_TYPEID::Concat:
         {
             const op::Concat* concat = static_cast<const op::Concat*>(&node);
             std::vector<const T*> in_args;
             std::vector<Shape> in_shapes;
-            for (std::shared_ptr<HostTensor> arg : args)
+            for (size_t i = 0; i < node.get_input_size(); i++)
             {
-                in_args.push_back(arg->get_data_ptr<T>());
-                in_shapes.push_back(arg->get_shape());
+                in_args.push_back(static_cast<const T*>(args[i]));
+                in_shapes.push_back(node.get_input_shape(i));
             }
             reference::concat<T>(in_args,
-                                 out[0]->get_data_ptr<T>(),
+                                 static_cast<T*>(out[0]),
                                  in_shapes,
-                                 out[0]->get_shape(),
+                                 node.get_output_shape(0),
                                  concat->get_concatenation_axis());
             break;
         }
         case OP_TYPEID::Constant:
         {
-            const op::Constant* c = static_cast<const op::Constant*>(&node);
-            reference::constant<T>(
```

Comment:
Maybe we should remove `reference::constant` at some point, if it's not used elsewhere.

Reply:
Sounds like a good idea. I think Constant is handled uniquely for each backend and none use reference::constant